### PR TITLE
graph-framework-update

### PR DESCRIPTION
**Pull request by cygnus-x-1**
:art: made ReadableGraph an AutoCloseable in order to support try-with-resources
:art: removed deprecated parts of the graph api
:art: deleted the unused GraphUtilities class
:art: moved Timer, IntHashSet and ImmutableObjectCache to the Utilities module
:art: added hundreds of finals

### CODE REVIEW

File: CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/ReadableGraph.java [hunk 1]
```diff
@@ -15,20 +15,23 @@
...
 
     /**
      * Releases the read lock on the graph. After this is called, this
      * ReadableGraph is now considered invalid and should not be used from that
      * point on.
      */
     public void release();
+    
+    @Override
+    public default void close() {
```
Comment (arcturus2) [inline CoreGraphFramework/src/au/gov/asd/tac/constellation/graph/ReadableGraph.java:34]: 😄

**Comment by arcturus2** [comment]
I ran the unit tests in a different branch and Im getting the failures there too so disregard my comment.

**Comment by cygnus-x-1** [comment]
> I ran the unit tests in a different branch and Im getting the failures there too so disregard my comment.

It looks like there are now many broken unit tests. I fixed one of them and it was to do with attribute ids changing as they are copied between graphs. This is expected behaviour (which the unit test wasn't handling), but it does imply that something has changed which makes it more likely for attribute ids to change. I suspect this might have been introduced when `StoreGraph` was altered recently.

File: CoreUtilities/nbproject/project.xml [hunk 2]
```diff
@@ -163,6 +163,7 @@
                 <package>au.gov.asd.tac.constellation.utilities.support</package>
                 <package>au.gov.asd.tac.constellation.utilities.temporal</package>
                 <package>au.gov.asd.tac.constellation.utilities.text</package>
+                <package>au.gov.asd.tac.constellation.utilities.time</package>
```
[the remark this answers is not in the document]
Comment (arcturus2) [reply CoreUtilities/nbproject/project.xml]: @cygnus-x-1 just letting you know about this comment in case you missed it.

**Pull request merged**